extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/catalog/CatalogApiService.java

```diff
@@ -34,13 +37,17 @@ public class CatalogApiService {
     private final DspCatalogService dspCatalogService;
 
     public List<UiDataOffer> fetchDataOffers(String connectorEndpoint) {
-        var dspCatalog = dspCatalogService.fetchDataOffers(connectorEndpoint);
-        var endpoint = dspCatalog.getEndpoint();
-        var participantId = dspCatalog.getParticipantId();
+        try {
+            var dspCatalog = dspCatalogService.fetchDataOffers(connectorEndpoint);
+            var endpoint = dspCatalog.getEndpoint();
+            var participantId = dspCatalog.getParticipantId();
 
-        return dspCatalog.getDataOffers().stream()
-                .map(dataOffer -> buildDataOffer(dataOffer, endpoint, participantId))
-                .toList();
+            return dspCatalog.getDataOffers().stream()
+                    .map(dataOffer -> buildDataOffer(dataOffer, endpoint, participantId))
+                    .toList();
+        } catch (DspCatalogServiceException e) {
+            throw new ServiceException(e, Response.Status.BAD_GATEWAY);
```

Comment:
would internal server error make more sense?

Reply:
This is an already existing exception that seems to represent a generic problem when using the service.
I'm not for InternalServerError because we didn't fail completely, we handled everything as expected. 500 would be worse, such as an exception that we didn't catch.
I don't see a org.eclipse.edc.web.spi.exception.BadGatewayException but it doesn't extend the correct class and so will not be processed to be returned as an HTTP502 (I think). Any exception that we use here should be a WebApplicationException.